modyn/trainer_server/internal/trainer/pytorch_trainer.py

```diff
@@ -263,26 +301,11 @@ def train(self) -> None:  # pylint: disable=too-many-locals, too-many-branches
                 except queue.Empty:
                     pass
 
-                sample_ids = batch[0]
-                if isinstance(sample_ids, torch.Tensor):
-                    sample_ids = sample_ids.tolist()
-                elif isinstance(sample_ids, tuple):
-                    sample_ids = list(sample_ids)
+                sample_ids, target, data = self.prepare_data(batch)
 
-                assert isinstance(sample_ids, list), "Cannot parse result from DataLoader"
-
-                if self._label_tranformer_function is None:
-                    target = batch[2].to(self._device)
-                else:
-                    target = self._label_tranformer_function(batch[2]).to(self._device)
-
-                data: Union[torch.Tensor, dict]
-                if isinstance(batch[1], torch.Tensor):
-                    data = batch[1].to(self._device)
-                elif isinstance(batch[1], dict):
-                    data: dict[str, torch.Tensor] = {}  # type: ignore[no-redef]
-                    for name, tensor in batch[1].items():
-                        data[name] = tensor.to(self._device)
+                if self._downsampling_mode == DownsamplingMode.SAMPLE_THEN_BATCH:
+                    weights = batch[3]
+                    weights = weights.float()
```

Comment:
sorry, why do we only fetch the weights when we do StB? what about no downsampling at all (e.g., in autonomous driving, we might assing weights at the selector that prioritize data points such as night time etc, that are more rare)